src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js

```diff
@@ -314,13 +316,16 @@ function ComposerWithSuggestions({
         [suggestionsRef],
     );
 
-    const updateShouldShowSuggestionMenuToFalse = useCallback(() => {
-        if (!suggestionsRef.current) {
+    const hideSuggestionMenu = useCallback(() => {
+        if (!suggestionsRef.current || shouldIgnoreScrollCallback.current) {
+            shouldIgnoreScrollCallback.current = false;
```

Comment:
I'm not sure whether this is a right way to lower this flag...

Wouldn't this be more direct?

```
    const debouncedLowerIsScrollLikelyLayoutTriggered = useDebounce(() => {
        isScrollLikelyLayoutTriggered.value = false;
    }, 200);
    
    const raiseIsScrollLikelyLayoutTriggered = () => {
        isScrollLikelyLayoutTriggered.value = true;
        debouncedLowerIsScrollLikelyLayoutTriggered();
    }
``

Then, we could call `raiseIsScrollLikelyLayoutTriggered` when the text changes. And check if `isScrollLikelyLayoutTriggered.value` directly in the `onScoll` callback.

Reply:
Also, I suggested using a `useDebounce` hook instead of directly calling `_.debounce` inside `useMemo`, because technically speaking starting a debounce is a side-effect...

I _think_ that a possible implementation could be:

```js
const useDebounce = (func, wait, options) => {
    const debouncedFnRef = useRef();

    useEffect(() => {
        const debouncedFn = _.debounce(func, wait, options);

        debouncedFnRef.current = debouncedFn;

        return () => debouncedFn.cancel();
    });

    return (...args) => {
        const debouncedFn = debouncedFnRef.current;

        if (debouncedFn) {
            debouncedFn(...args);
        }
    };
};
```

(Based on lodash `debounce`)

But this would need to be tested.

Reply:
> The useEffect does not have any deps

Yeah, I was writing it out of my head, it was just a starting point. You're right that deps are required here.

> The question is, are we okay to make the timeout 500ms?

You're talking about a Native build, I presume?

We're fine with a value that's good at distinguishing the layout-related scrolls from the manual ones, taking into account that a real user rarely scrolls _immediately_ after typing.

We should test this on a release build, preferably on a physical device, as in my experience it makes a difference. Compiling a release build is very slow, so you can create a spin-off branch that adds a debug UI component for choosing the `wait` value. Visually, it can be placed anywhere. Then we can experiment with this manually.

Reply:
> using the timeout that works on release mode means that the issue will still happen on debug and I think that's bad.

This is a tradeoff thing.

- A true positive is when we classify a scroll as layout-related and it actually was triggered by a layout.
- A false positive is when we classify a scroll as layout-related but actually, the user scrolled manually
- A true negative is when we classify a scroll as manual, and the user actually scrolled manually
- A false negative is when we classify a scroll as manual, when in fact it was triggered by a layout

If we make the time window wider to minimize false negatives on the debug build, we might increase the amount of false positives on the release build. So we should be careful here.

Still, I agree that it would be beneficial to at least ensure that his bug occurs _rarely_ on the debug builds.

Whatever we do, we have to test this on the release build too.

Reply:
> I think the false positive is not that bad in our case.

You might be right. Also, the user-triggered scroll will consist of multiple interactions in every typical case, so even if the first touch/wheel-scroll won't close the suggestion menu, the later one will.

Reply:
I think we are, I'm not against any particular number, as I mentioned. I will test it and figure out if I can see any UX issues with it. Because we are in agreement that the only thing this number does is draw the boundary between true/false positives (/negatives) on debug/release modes, right?

Also, the constant itself isn't the only open topic here. We also have the issue of impureness in `useMemo` and the wording of the comment we leave for the others.